README.md

```diff
@@ -45,19 +46,23 @@ grype <image> --scope all-layers
 Grype can scan a variety of sources beyond those found in Docker.
 
 ```
-# scan a docker image tar (from the result of "docker image save ... -o image.tar" command)
-grype docker-archive://path/to/image.tar
+# scan a container image archive (from the result of `docker image save ...`, `podman save ...`, or `skopeo copy` commands)
+grype path/to/image.tar
 
 # scan a directory
-grype dir://path/to/dir
+grype path/to/dir
```

Comment:
I understand this has probably been discussed a lot, but I thought these changes were to put some order in the usage of schemes, not generalize them to be able to infer the destination. Specifically this line in the in this PR's subject: 

>  TL;DR is :// is being replaced with : to fall in line with the URI RFC.

Automatically inferring is going to be problematic for integrations because it was the way to detect what the source type is (dir, tar, docker, etc..). If `grype` allows these inputs, then the integrations need to enforce a constraint, by either not allowing these free-styling of input, or separating input in the form of configuration.

That would mean that instead of these:

```yaml
    - name: Run directory Scan Action
      uses: anchore/scan-action@v1
      with:
        source: "dir://tests/python"

    - name: Run Image Scan Action
      uses: anchore/scan-action@v1
      with:
        source: "alpine:3.12"
```

The action would require prefixing the scheme or separating the options like this:

```yaml
    - name: Run directory Scan Action
      uses: anchore/scan-action@v1
      with:
        dir-source: "tests/python"

    - name: Run Image Scan Action
      uses: anchore/scan-action@v1
      with:
        img-source: "alpine:3.12"
```

Reply:
What prevents grype from inferring from generic input? That is, what prevents this:
```
    - name: Run directory Scan Action
      uses: anchore/scan-action@v1
      with:
        source: "tests/python"

    - name: Run Image Scan Action
      uses: anchore/scan-action@v1
      with:
        source: "alpine:3.12"
```
or more explicitly from the user:
```
    - name: Run directory Scan Action
      uses: anchore/scan-action@v1
      with:
        source: "dir:tests/python"

    - name: Run Image Scan Action
      uses: anchore/scan-action@v1
      with:
        source: "docker:alpine:3.12"
```
Which would be the same as grype options:
```
grype docker:alpine:3.12
grype alpine:3.12
grype dir:tests/python
grype tests/python
```
Is there something in particular with the integration that it requires knowledge of the source type?

Reply:
It isn't that there is something that prevents it. But the integration must know if the source is a path or an image for reporting.

Reply:
I think that implicit input from a user is OK here, as the user does not need to understand explicit schemes. Also this does not block extensions from enforcing explicit usage of schemes (either by prepending strings, string validation, or otherwise). I think your comments are quite notable, but mostly impact the integration implementation, not the CLI surface area on grype meant for the user.